istioctl/cmd/istioctl/kubeinject.go

```diff
@@ -255,35 +254,10 @@ istioctl kube-inject -f deployment.yaml -o deployment-injected.yaml --injectConf
 			// hub and tag params only work with ISTIOCTL_USE_BUILTIN_DEFAULTS
 			// so must be specified together. hub and tag no longer have defaults.
 			if hub != "" || tag != "" {
-				// ISTIOCTL_USE_BUILTIN_DEFAULTS is used to have legacy behaviour.
-				if !getBoolEnv("ISTIOCTL_USE_BUILTIN_DEFAULTS", false) {
-					return errors.New("one of injectConfigFile or injectConfigMapName is required\n" +
-						"use the following command to get the current injector file\n" +
-						"kubectl -n istio-system get configmap istio-sidecar-injector " +
-						"-o=jsonpath='{.data.config}' > /tmp/injectConfigFile.yaml")
-				}
-
-				if hub == "" || tag == "" {
-					return fmt.Errorf("hub and tag are both required. got hub: '%v', tag: '%v'", hub, tag)
-				}
-
-				if sidecarTemplate, err = inject.GenerateTemplateFromParams(&inject.Params{
-					InitImage:           inject.InitImageName(hub, tag, debugMode),
-					ProxyImage:          inject.ProxyImageName(hub, tag, debugMode),
-					Verbosity:           verbosity,
-					SidecarProxyUID:     sidecarProxyUID,
-					Version:             versionStr,
-					EnableCoreDump:      enableCoreDump,
-					Mesh:                meshConfig,
-					ImagePullPolicy:     imagePullPolicy,
-					IncludeIPRanges:     includeIPRanges,
-					ExcludeIPRanges:     excludeIPRanges,
-					IncludeInboundPorts: includeInboundPorts,
-					ExcludeInboundPorts: excludeInboundPorts,
-					DebugMode:           debugMode,
-				}); err != nil {
-					return err
-				}
+				return errors.New("one of injectConfigFile or injectConfigMapName is required\n" +
```

Comment:
We seem to still need it, the tests are doing the wrong thing and using the 'builtin defaults' ( which is the only place the hardcoded config should be used ). May be better to split into 2 PR, one adding the env and separate one (in master?) getting rid of the hack and fixing the tests.